### DISCUSSION TITLE
2023/03/27/leetcode%E5%88%B7%E9%A2%98%E8%AE%B0%E5%BD%955-%E5%AD%98%E5%9C%A8%E9%87%8D%E5%A4%8D%E5%85%83%E7%B4%A0/

### DISCUSSION THREAD

**Discussion by giscus** [Zhuanjier/blog#14] [Announcements]
# 2023/03/27/leetcode%E5%88%B7%E9%A2%98%E8%AE%B0%E5%BD%955-%E5%AD%98%E5%9C%A8%E9%87%8D%E5%A4%8D%E5%85%83%E7%B4%A0/

¶存在重复元素  原题传送门  给你一个整数数组 nums 。如果任一值在数组中出现 至少两次 ，返回 true ；如果数组中每个元素互不相同，返回 false 。 ¶示例 ¶示例1 输入：nums = [1,2,3,1] 输出：true ¶示例2 输入：nums = [1,2,3,4] 输出：false ¶示例3 输入：nums = [1,1,1,3,3,4,3,2,4,2] 输出：true 提

https://blog.zhuanjie.ltd/2023/03/27/leetcode%E5%88%B7%E9%A2%98%E8%AE%B0%E5%BD%955-%E5%AD%98%E5%9C%A8%E9%87%8D%E5%A4%8D%E5%85%83%E7%B4%A0/

**Comment by RandallAnjie** [comment]
在C语言中，unique函数用于去除一个数组中相邻的重复元素，其具体执行过程如下：

接收一个指向数组首元素的指针和数组中元素的个数作为参数。
从数组的第二个元素开始，依次比较当前元素和前一个元素是否相等。
如果相等，则将当前元素移动到数组的末尾，并将数组中元素的个数减一。
如果不相等，则继续比较下一个元素，直到比较完所有元素。
返回去重后的元素个数。
例如，对于以下数组：

int arr[] = {1, 2, 2, 3, 3, 3, 4, 5, 5};

调用unique函数后，数组变成：

int arr[] = {1, 2, 3, 4, 5, 3, 4, 5, 5};

去重后的元素个数为5。
